pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -115,6 +118,14 @@ public synchronized void init(PinotConfiguration config, HelixManager helixManag
       throws Exception {
     LOGGER.info("Initializing Helix instance data manager");
 
+    boolean caseSensitive = !config.getProperty(
+        CommonConstants.Helix.ENABLE_CASE_INSENSITIVE_KEY,
+        CommonConstants.Helix.DEFAULT_ENABLE_CASE_INSENSITIVE);
+    if (caseSensitive) {
+      _tableDataManagerMap = new ConcurrentHashMap<>();
+    } else {
+      _tableDataManagerMap = Collections.synchronizedMap(new CaseInsensitiveMap<>());
```

Comment:
This changes semantics and performance. Couldn't we use a normal ConcurrentHashMap and if needed (see Jackie's comment above) change the way we use the map to run a case-insensitive lookup?

Reply:
Yeah, I followed Jackie's advice and found a way to do that only for the MSE path. Apparently it is enough will resolving actual table names while creating the `DispatchablePlan`. Hope that's more aligned with what we are looking for.